cassiopeia/ProcessingPipeline/process/filter_utils.py

```diff
@@ -0,0 +1,219 @@
+import logging
+import os
+import sys
+
+import Levenshtein
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import time
+from tqdm import tqdm
+
+from . import lineageGroup_utils as lg_utils
+
+
+def record_stats(moleculetable: pd.DataFrame) -> (np.array, np.array, np.array):
+    """
+    Simple function to record the number of UMIs.
+
+    Args:
+        moleculetable: A DataFrame of alignments
+
+    Returns:
+        Read counts for each alignment, number of unique UMIs per intBC, number
+            of UMIs per cellBC
+    """
+
+    # Count UMI per intBC
+    umi_per_ibc = np.array([])
+    for n, g in tqdm(moleculetable.groupby(["cellBC"]), desc="Recording stats"):
+        x = g.groupby(["intBC"]).agg({"UMI": "nunique"})["UMI"]
+        if x.shape[0] > 0:
+            umi_per_ibc = np.concatenate([umi_per_ibc, np.array(x)])
+
+    # Count UMI per cellBC
+    umi_per_cbc = (
+        moleculetable.groupby(["cellBC"])
+        .agg({"UMI": "count"})
+        .sort_values("UMI", ascending=False)["UMI"]
+    )
+
+    return (
+        np.array(moleculetable["ReadCount"]),
+        umi_per_ibc,
+        np.array(umi_per_cbc),
+    )
+
+
+def filter_cell_bcs(
+    moleculetable: pd.DataFrame, umiCountThresh: int = 10, verbose: bool = False
+):
+    """
+    Filter out cell barcodes that have too few UMIs.
+
+    Filters out cell barcodes that have a number of unique UMIs <= umiCountThresh.
+    Assumes that UMIs have been resolved, i.e. that each UMI only appears once.
+
+    Args:
+        moleculetable: A DataFrame of alignments to be filtered
+        umiCountThresh: The minimum number of UMIs needed for a cellBC to be
+            included in the filtered DataFrame
+        verbose: Indicates whether to log the number of cellBCs and UMIs
+            remaining after filtering
+
+    Returns:
+        n_moleculetable: A filtered DataFrame of alignments.
+    """
+
+    # Create a cell-filter dictionary for hash lookup lmter on when filling
+    # in the table
+    cell_filter = {}
+
+    for n, group in tqdm(
+        moleculetable.groupby(["cellBC"]), desc="Filter cells"
+    ):
+        if group.shape[0] <= umiCountThresh:
+            cell_filter[n] = "bad"
+            # tooFewUMI_UMI.append(group.shape[0])
+        else:
+            cell_filter[n] = "good"
+
+    # apply the filter using the hash table created above
+    moleculetable["status"] = moleculetable["cellBC"].map(cell_filter)
+
+    # filter based on status & reindex
+    n_moleculetable = moleculetable[(moleculetable["status"] == "good")]
+    n_moleculetable.index = [i for i in range(n_moleculetable.shape[0])]
+
+    if verbose:
+        lg_utils.generate_log_output(n_moleculetable)
+
+    return n_moleculetable
+
```

Comment:
We already have this function in `utilities.py`. Can you remove this function and just make calls to `cassiopeia.pp.filter_cellBCs`?